src/Native/Unix/System.Security.Cryptography.Native/openssl.cpp

```diff
@@ -793,7 +794,7 @@ static int CheckX509HostnameMatch(ASN1_STRING* candidate, const char* hostname,
         return 0;
     }
 
-    return !memcmp(candidate->data, hostname, static_cast<size_t>(cchHostname));
+    return !strncasecmp(reinterpret_cast<char*>(candidate->data), hostname, static_cast<size_t>(cchHostname));
```

Comment:
`unsigned char*` => `char*` (the C version cast it directly to `const char*`, but one lets the const-implicit happen)